### ISSUE TITLE
WoW 10.0 -  counter display bug

### ISSUE THREAD

**Issue by Voxxel** [Choonster-WoW-AddOns/YarkoCooldowns#1]
I've read you're not willing to update this addon because of OmniCC, but this is not the same thing, YarkoCooldowns are unique and far superior than OmniCC at least in the counter displaying style. Due to it's unique color switching method close to the cooldown's expiration time this addon is still a must for me. It's catching the eyes immediately unlike OmniCC's cooldown counter.

With 10.0 there's a returning bug where the counter shows up on various elements and panels of the UI where it shouldn't be. This bug is similar to the Shadowlands bug (screenshot below) but this time new Dragonflight UI panels included as well.
 
![hRUb1O5](https://user-images.githubusercontent.com/3882152/199202802-60bb4b97-8051-4024-99f4-4a5e1c5f8ed5.jpg)

**Comment by Choonster** [comment]
I don't play WoW any more, so I don't really know where it's showing that it shouldn't be.

With 4.5, I've gone through the default UI source code and reviewed every `<Cooldown>` widget created in XML to see whether it's actually a cooldown or something like a progress bar where the text shouldn't be displayed; so the Major Faction Renown and Profession Specialization progress bars are now ignored.

If there are any elements I've missed, please send me a screenshot and ideally the full name of the element or panel (from `/fstack`) so I know what to add to the ignore list.

**Comment by Voxxel** [comment]
In 4.4.1 I found only 3 instances related to DF so far, those two you mentioned above and the Dragon Isles Summary page (ExpansionLandingPage).
However the Profession Specialization and the Dragon Isles Summary pages are only partially fixed in 4.5. :
 
![landing-441](https://user-images.githubusercontent.com/3882152/200167291-16a0d41c-7e98-4d94-beb8-c2e2182ab299.jpg)
![landing-45](https://user-images.githubusercontent.com/3882152/200167289-142c524d-9937-46ec-8683-fd4707793c69.jpg)
 
![prof-441](https://user-images.githubusercontent.com/3882152/200167305-493d6621-52a9-405f-a90e-ed21797eddce.jpg)
![prof-45](https://user-images.githubusercontent.com/3882152/200167304-77c7c238-5c1e-40a6-bb36-e2011cc77c56.jpg)

**Comment by Voxxel** [comment]
Are these fstack element names are proper to exclude or do you need to point over other names?

**Comment by Choonster** [comment]
Those were helpful, thanks. It looks like I wasn't actually calling the function to ignore these UI elements when the AddOns providing them were loaded.

Can you see if 4.5.1 fixes this?

**Comment by Voxxel** [comment]
Hey, 
Thank you for the fix. The landing page counter is now gone with 4.5.1 but the small counter of the profession specialization panels are still there. 
 
I don't know if they changed something since the last build so here are some new screenshots:

 
![WoWScrnShot_111222_104736](https://user-images.githubusercontent.com/3882152/201468853-fc43a6b8-6413-4520-b8bb-e75a15343bc5.jpg)
![WoWScrnShot_111222_104658](https://user-images.githubusercontent.com/3882152/201468854-258e95eb-3e4e-4d8c-878c-6ecccbb79757.jpg)
![WoWScrnShot_111222_104709](https://user-images.githubusercontent.com/3882152/201468856-2d961fd3-a882-4056-b70b-20ccd2dd60b2.jpg)
![WoWScrnShot_111222_104723](https://user-images.githubusercontent.com/3882152/201468858-8a8a23b2-da0d-4f09-b96a-7f89a85a38d4.jpg)

**Comment by Choonster** [comment]
4.5.2 should ignore the Profession Trait progress bars now. Please let me know if there are any more issues.

**Comment by Voxxel** [comment]
It's still showing the counter on Profession Specialization traits even with 4.5.2.

I wanted to check if I installed the new version correctly but in the addon list it just list Cooldowns @project version@ : 
 
![WoWScrnShot_111522_174225](https://user-images.githubusercontent.com/3882152/201983932-b25c0b2a-a9a1-4ba2-870f-23a46d5c97b7.jpg)

And it's not listed in the addon's settings either. So I checked the addon directory and it looks like it's the new v4.5.2 :

![ycd-ver](https://user-images.githubusercontent.com/3882152/201984131-44792352-9ee4-485b-83fa-1fd4a3286371.png)

So I packed up some more screenshots with different elements listing in /fstack: 
[Screenshots.zip](https://github.com/Choonster-WoW-AddOns/YarkoCooldowns/files/10014816/Screenshots.zip)

**Comment by Choonster** [comment]
You can download specific versions from the [Releases page](https://github.com/Choonster-WoW-AddOns/YarkoCooldowns/releases), though these are just packaged forms of the code that's in the repo.

I've just made a change so that existing Profession Trait progress bars are also ignored, not just new ones; though I haven't released this as a version. Could you see if this fixes it? Also let me know if it prints `Disabling for existing Profession Spec Talent Button` or `Disabling for new Profession Spec Talent Button` or a combination of the two in the chat.

**Comment by Voxxel** [comment]
This new version  d63f9cd  is also showing the counter for traits as before. I realized the counters not always pop up on the traits they remain hidden rarely while I'm switching between tabs on the panel but 90% of times the counters are up. I don't know if it has to do anything with this lua bug or not so this may cause because of a blizzard bug:

```
1x Usage: local entryInfo = C_Traits.GetEntryInfo(configID, entryID)
[string "=[C]"]: ?
[string "@Blizzard_SharedTalentUI/Blizzard_SharedTalentUtil.lua"]:265: in function `GetEntryInfo'
[string "@Blizzard_Professions/Blizzard_ProfessionsSpecializationsTemplates.lua"]:274: in function `GetRanks'
[string "@Blizzard_Professions/Blizzard_ProfessionsSpecializationsTemplates.lua"]:384: in function `AddTooltipInfo'
[string "@Blizzard_Professions/Blizzard_ProfessionsSpecializationsTemplates.lua"]:310: in function <...ons/Blizzard_ProfessionsSpecializationsTemplates.lua:300>
[string "=[C]"]: ?

Locals:
(*temporary) = "Usage: local entryInfo = C_Traits.GetEntryInfo(configID, entryID)"
```


I also get these 2 types of counting lua errors related to Cooldowns when I'm opening/closing the professions panels / spending points on traits / etc :


```
87x Cooldowns/Cooldowns.lua:157: attempt to index local 'talentButton' (a number value)
[string "@Cooldowns/Cooldowns.lua"]:157: in function <Cooldowns/Cooldowns.lua:156>
[string "=[C]"]: ?
[string "@SharedXML/CallbackRegistry.lua"]:143: in function <SharedXML/CallbackRegistry.lua:142>
[string "=[C]"]: ?
[string "@SharedXML/CallbackRegistry.lua"]:146: in function `TriggerEvent'
[string "@Blizzard_SharedTalentUI/Blizzard_SharedTalentFrame.lua"]:726: in function `ReleaseAllTalentButtons'
[string "@Blizzard_Professions/Blizzard_ProfessionsSpecializations.lua"]:471: in function `LoadTalentTreeInternal'
[string "@Blizzard_SharedTalentUI/Blizzard_SharedTalentFrame.lua"]:1024: in function `LoadTalentTree'
[string "@Blizzard_SharedTalentUI/Blizzard_SharedTalentFrame.lua"]:249: in function `OnShow'
[string "@Blizzard_Professions/Blizzard_ProfessionsSpecializations.lua"]:163: in function <..._Professions/Blizzard_ProfessionsSpecializations.lua:162>
[string "=[C]"]: in function `Show'
[string "@FrameXML/UIParent.lua"]:2853: in function `SetUIPanel'
...
[string "=[C]"]: in function `UseAction'
[string "@FrameXML/SecureTemplates.lua"]:364: in function `handler'
[string "@FrameXML/SecureTemplates.lua"]:690: in function <FrameXML/SecureTemplates.lua:672>
[string "@FrameXML/SecureTemplates.lua"]:704: in function <FrameXML/SecureTemplates.lua:697>
[string "@FrameXML/SecureTemplates.lua"]:746: in function <FrameXML/SecureTemplates.lua:730>
[string "=[C]"]: ?
[string "@FrameXML/SecureHandlers.lua"]:266: in function <FrameXML/SecureHandlers.lua:263>
[string "=[C]"]: ?
[string "@FrameXML/SecureHandlers.lua"]:296: in function <FrameXML/SecureHandlers.lua:279>
[string "=(tail call)"]: ?

Locals:
talentButton = 99
(*temporary) = nil
(*temporary) = "attempt to index local 'talentButton' (a number value)"
```
 ```

75x Cooldowns/Cooldowns.lua:152: attempt to index local 'talentButton' (a number value)
[string "@Cooldowns/Cooldowns.lua"]:152: in function <Cooldowns/Cooldowns.lua:151>
[string "=[C]"]: ?
[string "@SharedXML/CallbackRegistry.lua"]:143: in function <SharedXML/CallbackRegistry.lua:142>
[string "=[C]"]: ?
[string "@SharedXML/CallbackRegistry.lua"]:146: in function `TriggerEvent'
[string "@Blizzard_SharedTalentUI/Blizzard_SharedTalentFrame.lua"]:509: in function `AcquireTalentButton'
[string "@Blizzard_Professions/Blizzard_ProfessionsSpecializations.lua"]:378: in function `InstantiateTalentButton'
[string "@Blizzard_Professions/Blizzard_ProfessionsSpecializations.lua"]:528: in function `LoadTalentTreeInternal'
[string "@Blizzard_SharedTalentUI/Blizzard_SharedTalentFrame.lua"]:1024: in function `LoadTalentTree'
[string "@Blizzard_SharedTalentUI/Blizzard_SharedTalentFrame.lua"]:249: in function `OnShow'
[string "@Blizzard_Professions/Blizzard_ProfessionsSpecializations.lua"]:163: in function <..._Professions/Blizzard_ProfessionsSpecializations.lua:162>
[string "=[C]"]: in function `Show'
...
[string "=[C]"]: in function `UseAction'
[string "@FrameXML/SecureTemplates.lua"]:364: in function `handler'
[string "@FrameXML/SecureTemplates.lua"]:690: in function <FrameXML/SecureTemplates.lua:672>
[string "@FrameXML/SecureTemplates.lua"]:704: in function <FrameXML/SecureTemplates.lua:697>
[string "@FrameXML/SecureTemplates.lua"]:746: in function <FrameXML/SecureTemplates.lua:730>
[string "=[C]"]: ?
[string "@FrameXML/SecureHandlers.lua"]:266: in function <FrameXML/SecureHandlers.lua:263>
[string "=[C]"]: ?
[string "@FrameXML/SecureHandlers.lua"]:296: in function <FrameXML/SecureHandlers.lua:279>
[string "=(tail call)"]: ?

Locals:
talentButton = 98
(*temporary) = <function> defined @FrameXML/RestrictedInfrastructure.lua:124
(*temporary) = "YarkoCooldowns"
(*temporary) = "Disabling for new Profession Spec Talent Button"
(*temporary) = nil
(*temporary) = 98
(*temporary) = "attempt to index local 'talentButton' (a number value)"
```

![WoWScrnShot_112022_100941](https://user-images.githubusercontent.com/3882152/202894202-1f884407-dfc0-47e9-ba79-191dd39e3e46.jpg)

**Comment by Choonster** [comment]
Those errors should be fixed by c57d40591dcc81ac74439ede338c7890ca66213a, can you please try that version?

**Comment by Voxxel** [comment]
Yes, it's good now aside it prints out a lot of white text about hiding counters in the chat.

**Comment by Choonster** [comment]
I've just released [4.5.3](https://github.com/Choonster-WoW-AddOns/YarkoCooldowns/releases/tag/4.5.3) which contains the previous fixes and also removes the debug printing.

Thanks for the help debugging this, and let me know if there are any other elements that need to be ignored.

**Comment by Voxxel** [comment]
Thank you!
